docs/guides/collections.rst

```diff
@@ -1,105 +1,144 @@
+of the information if would normally return using the ``all()`` or indeed other methods. To filter a
+collection object, use the following pattern
 
 .. code-block:: python
 
-                from cfme.utils.appliance import BaseCollection, BaseEntity
+                dc = appliance.collection.datastore
+                dc_filtered = dc.filter({'provider': provider})
 
-                class ObjectCollection(BaseCollection):
-                    """An object collection"""
+Automatically generated filtered collections
+--------------------------------------------
 
-                    def __init__(self, appliance):
-                        self.appliance = appliance
+``BaseEntity`` objects have the special ability to create filtered collections. These appear, much
+like the ``collections`` attribute on the ``IPAppliance`` instances. Consider the example above where
+the ``Repository`` object is given the ``_collections`` attribute. This contains a dictionary of
+collection names, along with the collection class that should be instantiated. The collection is then
+instantiated with the following filter, like so:
 
-                    def instantiate(self, name, label):
-                        return Object(self, name, label)
+.. code-block:: python
 
-                    def create(self, name, label):
-                        run_create(name, label)
-                        return self.instantiate(name, label)
+                repo = appliance.collections.repositories.all()[0]
+                playbook_collection = repo.collections.playbooks
+                playbook_collection.all()  # returns ONLY playbooks from that repo
 
-                class Object(BaseEntity):
-                    """An object entity"""
+                # equivalent code
+                playbook_collection = PlaybookCollection(self.appliance, filters={'parent': self})
 
-                    def __init__(self, collection, name, label):
-                        self.collection = collection
-                        self.appliance = self.collection.appliance
+In the example above, the ``BaseEntity`` automatically instantiates the playbook collection object
+with a ``parent`` filter. The playbook collection object would then need to honour that filter
+when returning the playbooks. A collection isn't under any obligation to support a certain filter.
 
-                    def update(self, *updates):
-                        some_update_mechanism(*updates)
+.. note:: In the future filter names which are supported may need to be defined somewhere to allow
+          unsupported filters to be reported as warnings.
+
+Collection Methods
+------------------
+
+* ``__init__()`` - This method is hidden inside the ``BaseCollection`` object and shouldn't be
+  overidden without good reason. There are exceptional circumstances. Mainly if work **MUST** be
+  done at init time, using ``__attrs_post_init__`` method is used instead.
```

Comment:
**REQUIRED**  at this point we shouldn't point people to how attrs work, we should point them to us to discuss, simply to avoid having to send them back after coming up with such a  change - quich very easyly could be something that can be solved more nicely

Reply:
Agreed